packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-12/pre.test.js

```diff
@@ -0,0 +1,15 @@
+import test from 'ava';
+
+import { waitForBlock } from '../coreUpgradeHelpers.mjs';
+import { main } from './tools/vat-status.mjs';
+
+test.before(async () => {
+  console.log('Wait for upgrade to settle');
```

Comment:
should the actions of the preceding upgrade be responsible for this?

Reply:
what I mean is, the thing that is "performing the actual upgrade" has the job of upgrading. if the next step has to wait for the upgrade to settle, the the upgrade step hasn't finished it's job. The test should be able to assume the upgrade is complete. Knowing how much to wait for it to complete should be the responsibility of the upgrade, since it depends on the upgrade being performed.